src/common/transformations/src/ov_ops/rotary_positional_embeddings.cpp

```diff
@@ -45,13 +45,23 @@ void RoPE::validate_and_infer_types() {
     }
 
     if (m_config.is_chatglm) {
-        // chatGLM specific RoPE
-        // input  [length, batch_size, (hidden_states_q + hidden_states_k + hidden_states_v)]
-        // output [length, batch_size, head_cnt, hidden_states_k]
-        set_output_type(
-            0,
-            get_input_element_type(0),
-            {input_pshape[0], input_pshape[1], ov::Dimension(m_config.head_cnt), ov::Dimension(m_config.head_size)});
+        if (m_config.support_2d_rope) {
```

Comment:
@ahnyoung-paul Do you test the fusion with latest glm-4 IR model, e.g. WW41? I find that the pattern is changed and the fusion is not working.

Reply:
I will check it. thanks.

Reply:
fixed the failure for cpu plugin 2d rope fusion matching.